notascript/src/App.js

```diff
@@ -118,21 +129,32 @@ export default class App extends Component {
 								<Overview
 									documentList={this.state.documents}
 									{...props}
-									onDelete={document => this.deleteDocumentFromList(document)}
+									onDelete={document => this.deleteDocument(document)}
 								/>
 							)}
 						/>
 						<Route
 							path='/details/:title'
 							render={props => (
 								<DocumentDetailView
-									selectedDocument={this.selectDocumentForDetailView({ props })}
+									selectedDocument={this.showDetails({ props })}
 									documents={this.state.documents}
 									{...props}
```

Comment:
I think if you don't need history, location or match in your DocumentDetailView you don't need to hand over {...props} to this component